### PR TITLE
feat: [IDP-3578]: Enhanced the RBAC docs

### PR DESCRIPTION
**Pull request by Debanitrkl**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/internal-developer-portal/rbac/resources-roles.md [hunk 1]
```diff
@@ -6,7 +6,66 @@ sidebar_position: 1
...
+  description: Description of my new service
+  tags:
+    - hidden
+spec:
+  type: service
+  lifecycle: production
+  owner: team-a
+```
+
+## RBAC in Workflows
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/rbac/resources-roles.md:51]:
```suggestion
## Access Control for Self Service Workflows
```

File: docs/internal-developer-portal/rbac/resources-roles.md [hunk 2]
```diff
@@ -6,7 +6,66 @@ sidebar_position: 1
...
+
+### View Access
+
+Every user in the account will have view access to the workflows.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/rbac/resources-roles.md:55]:
```suggestion
Every user in the account has view access to all the Workflows in IDP. There is an upcoming [feature request](https://developer.harness.io/roadmap/#idp) to remove view access if the user does not have view access of the pipelines powering up the Workflow.
```

File: docs/internal-developer-portal/rbac/resources-roles.md [hunk 3]
```diff
@@ -6,7 +6,66 @@ sidebar_position: 1
...
+
+### Edit and Run Access
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/rbac/resources-roles.md:57]: @Debanitrkl need to split Edit access and Execute separately. Edit is again done at the YAML level. Use the same texts from Catalog Edit above.

Comment (Debanitrkl) [reply docs/internal-developer-portal/rbac/resources-roles.md:57]: Sure let me separate that out

**Review by OrkoHunter — APPROVED** [review]

**Pull request merged**